src/common/blob/BlobClient.js

```diff
@@ -258,7 +258,12 @@ define(['blob/Artifact', 'blob/BlobMetadata', 'superagent'], function (Artifact,
                     var response = req.xhr.response; // response is an arraybuffer
                     if (contentType === 'application/json') {
                         var utf8ArrayToString = function (uintArray) {
-                            return decodeURIComponent(escape(String.fromCharCode.apply(null, uintArray)));
+                            var inputString = '',
+                                i;
+                            for (i = 0; i < uintArray.byteLength; i++) {
+                                inputString += String.fromCharCode(uintArray[i]);
+                            }
+                            return decodeURIComponent(escape(inputString));
                         };
```

Comment:
Funny you should ask, I made this:
http://jsperf.com/arr-fromcharcode

By changing the chunk size, I was able to make all of the tests run with different timing. I could not make it produce reliable results